@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1980,6 +1980,7 @@ export class TextElementsRenderer {
 
             // If the text is not visible nor optional, we won't render the icon neither.
             else if (!renderIcon || !textIsOptional) {
+                renderIcon = false;
```

Comment:
why not keep renderIcon as const and add && !textIsOptional at the end of the condition?

Reply:
Resolving because this code will anyway be refactored later.